src/mqtt_client/mod.rs

```diff
@@ -21,7 +21,7 @@ const MAX_RECURSION_DEPTH: usize = 8;
 // republished.
 const REPUBLISH_TIMEOUT_SECONDS: u32 = 2;
 
-type Iter<M> = crate::PathIter<'static, M, MAX_RECURSION_DEPTH, String<MAX_TOPIC_LENGTH>>;
+type Iter<M> = crate::PathIter<'static, M, MAX_RECURSION_DEPTH, String<MAX_TOPIC_LENGTH>, 1>;
```

Comment:
I'd say that ideally all of these excess `1` params could be wiped with generic defaults.

Reply:
That one can't be remove since it's on a type, not a trait. But there were a bunch of others.